src/src_main/crypto_tc.c

```diff
@@ -257,14 +272,41 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t* p_in_frame, const uint16_t in_fra
             // TODO: If ARSN is transmitted in the SHSNF field (as in CMAC... don't double count those bytes)
             *p_enc_frame_len = temp_tc_header.fl + 1 + 2 + sa_ptr->shivf_len + sa_ptr->shsnf_len + sa_ptr->shplf_len +
                                TC_PAD_SIZE + sa_ptr->stmacf_len;
-            new_enc_frame_header_field_length = (*p_enc_frame_len) - 1;
+            new_enc_frame_header_field_length = (*p_enc_frame_len) - 1;            
             break;
         case SA_ENCRYPTION:
             // Ingest length + spi_index (2) + shivf_len (varies) + shsnf_len (varies)
             //   + shplf_len + arsn_len + pad_size
             *p_enc_frame_len = temp_tc_header.fl + 1 + 2 + sa_ptr->shivf_len + sa_ptr->shsnf_len + sa_ptr->shplf_len +
-                               sa_ptr->arsn_len + TC_PAD_SIZE;
+                               sa_ptr->arsn_len + TC_PAD_SIZE; //should point to shplf_len
             new_enc_frame_header_field_length = (*p_enc_frame_len) - 1;
+            
+            tf_payload_len = temp_tc_header.fl - TC_FRAME_HEADER_SIZE - segment_hdr_len - fecf_len + 1;
```

Comment:
tf_payload_len is already defined above (exactly the same as this).

Reply:
Good catch.  I think this is leftover as well, and was missed, and can go away.

Reply:
Removed

Reply:
Confirmed removed in commit 7b13732